core/src/main/java/org/axonframework/serialization/json/JacksonSerializer.java

```diff
@@ -204,26 +194,30 @@ protected ObjectReader getReader(Class<?> type) {
     @Override
     public <S, T> T deserialize(SerializedObject<S> serializedObject) {
```

Comment:
I would expect a new method added to `Serializer` like `<S, T> T deserialize(SerializedObject<S> serializedObject, Class<T> desiredType)`.

Reply:
Along with this idea, it would be beneficial to have some kind of mapping registry saying which types are mapped to which types, so we can automatically deserialize the message. If such mapping is not found, returning `UnknownSerializedType` is just fine.

Reply:
Some form of `AbstractSerializer` which the others extend from to settle this `UnknownSerializedType` would make sense to me too.

Reply:
Adding the `Class<T> desiredType` will not work, because you won't be able to return anything else than `T`. I'm not sure I understand what the mapping strategy would solve.

Reply:
I think the assumption is it might safe time.

Reply:
Imagine a case where you don't have a fully qualified class name on your classpath (it's a message from some other system) but you have something structure which you can use to deserialize the incoming message to. Providing `desiredType` would deserialize incoming message to the structure known to your module.

Reply:
Having a registry would do this thing automatically: if you don't have a type on your classpath it would search whether there is a mapping in the registry "com.example.someTypeIDontKnow" -> com.foo.ThisIsSimilarStructure.class.